### PR TITLE
Make downsample target index replicas configurable

### PR DESCRIPTION
**Pull request by salvatore-campagna**
Introducing this setting in the downsampling logic will allow us to override its value
for Serverless deployments by means of a plugin. In Steteful deployments we will
use a default value of 0 to avoid unnecessary replication. In Serverless deployments
a minimum value of 1 is required to set the value of `index.number_of_replicas`.
This is because, in Serverless deployments, indices without replicas are not readable
and that would break our persistent task recovery mechanism which relies on
reading the latest written tsid into the downsampling target index. Not having replicas
in Serverless would result in downsampling persistent tasks to always start from scratch.

### CODE REVIEW

**Review by martijnvg — APPROVED** [review]
Left a question. LGTM otherwise.

File: x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java [hunk 1]
```diff
@@ -728,10 +736,11 @@ private void createDownsampleIndex(
          * We should note that there is a risk of losing a node during the downsample process. In this
          * case downsample will fail.
          */
+        int numberOfReplicas = Integer.parseInt(settings.get(Downsample.DOWNSAMPLE_MIN_NUMBER_OF_REPLICAS, "0"));
```
Comment (andreidan) [inline x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java:739]: I believe we can use `settings.getAsInt` here

**Review by andreidan — APPROVED** [review]
LGTM, thanks Salvatore

File: x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/Downsample.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ public class Downsample extends Plugin implements ActionPlugin, PersistentTaskPl
 
     public static final String DOWSAMPLE_TASK_THREAD_POOL_NAME = "downsample_indexing";
     private static final int DOWNSAMPLE_TASK_THREAD_POOL_QUEUE_SIZE = 256;
+    public static final String DOWNSAMPLE_MIN_NUMBER_OF_REPLICAS = "downsample.min_number_of_replicas";
```
Comment (martijnvg) [inline x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/Downsample.java:48]: Is this the right place for this constant? Are we able to reference to this in serverless

Comment (salvatore-campagna) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/Downsample.java:48]: My understanding is that this is not important. As you see here it is just a string that happens to identify a Setting which only exists in Serverless deployments. So for Stateful deployments that string will result in no setting being found and the default value being used.

Comment (martijnvg) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/Downsample.java:48]: ok 👍

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine test this please

**Pull request merged**